distributed/client.py

```diff
@@ -1479,6 +1513,7 @@ async def _close(self, fast=False):
             return
 
         self.status = "closing"
+        _del_global_client(self)
```

Comment:
might be worth avoiding the lock acquisition if we didn't _set_as_default

```suggestion
        if self._set_as_default:
            _del_global_client(self)
```